utility/drivers.cpp

```diff
@@ -199,78 +159,190 @@ void l3gd20h_getRawTemperature(int8_t *pRawTemperature)
+    //   FS = 0b00111000 (+/- 16 g full scale; high resolution enable) = 0x38
+    switch (gGain) {
+      case LSM303_ACCEL_GAIN2G:
+        gain = 0x08;
+        break;
+      case LSM303_ACCEL_GAIN4G:
+        gain = 0x18;
+        break;
+      case LSM303_ACCEL_GAIN6G:
+        _lsm303_d_accel_config.gain = LSM303_ACCEL_GAIN8G;
+      case LSM303_ACCEL_GAIN8G:
+        gain = 0x28;
+        break;
+      case LSM303_ACCEL_GAIN16G:
+        gain = 0x38;
+        break;
+    }
+    lsm.writeAccReg(LSM303::CTRL_REG4_A, gain);
+
+    // 0x47 = 0b01000111
+    // ODR = 0100 (50 Hz ODR); LPen = 0 (normal mode); Zen = Yen = Xen = 1 (all axes enabled)
+    lsm.writeAccReg(LSM303::CTRL_REG1_A, 0x47);
+  }
+}
+
+// LSM303 Magnetometer Configurations
+void _lsm303_mag_config(lsm303_mag_scale_e gaussScale) {
+  uint8_t scale;
+  _lsm303_d_mag_config.scale = gaussScale;
+
+  // LSM303 D (Spaceboard variant)
   if (lsm.getDeviceType() == LSM303::device_D) {
-    lsm.writeReg(LSM303::CTRL6, LSM303_MAG_SCALE4GAUSS);
-    _lsm303_d_mag_config.scale = LSM303_MAG_SCALE4GAUSS;
-    // Ctrl register for magnetic resolution, data rate, temp enable
-    // TEMP_EN M_RES1 M_RES0 M_ODR2 M_ODR1 M_ODR0 LIR2 LIR1
-    //    0/1    1      1      1      1      0      0   0
-    // 0xF8 = 0b11111000 - temp enable, mag high res, defaults ODR, LIR
-    // enable onboard temp sensor
-    lsm.writeReg(LSM303::CTRL5, 0xf8);
-  } else {
-    // Ctrl register for mag scaling
-    // 0b001 +-1.3
-    // 0b010 +-1.9
-    // 0b011 +-2.5
-    // 0b100 +-4.0
-    // 0b101 +-4.7
-    // 0b110 +-5.6
-    // ob111 +-8.1
-    // GN2 GN1 GN0  0 -->
-    // 0b10000000 = 0x80 (+/- 4 gauss)
-    lsm.writeMagReg(LSM303::CRB_REG_M, 0x80);
-    _lsm303_d_mag_config.scale = LSM303_MAG_SCALE4GAUSS;
-
-    // Ctrl register for Data Output Rate & Temp sensor
-    // TEMP_EN 0 0 DO2 DO1 DO0 0 0
-    //    1    0 0  1   0   0  0 0
-    // 0x90 = 0b10010000 - temp enable, default data rate
-    lsm.writeMagReg(LSM303::CRA_REG_M, 0x90);
-  }
-}
-
-boolean lsm303_accel_init() {
+    // 0x64 = 0b01100100
+    // M_RES = 11 (high resolution mode); M_ODR = 001 (6.25 Hz ODR)
+    lsm.writeReg(LSM303::CTRL5, 0x64);
+
+    // Adjustable gauss scale:
+    //   MFS = 0b00000000 (+/-  2 gauss full scale) = 0x00
+    //   MFS = 0b00100000 (+/-  4 gauss full scale) = 0x20 (Default)
+    //   MFS = 0b01000000 (+/-  8 gauss full scale) = 0x40
+    //   MFS = 0b01100000 (+/- 12 gauss full scale) = 0x60
+    switch (gaussScale) {
+      case LSM303_MAG_SCALE1_3GAUSS:
+        _lsm303_d_mag_config.scale = LSM303_MAG_SCALE2GAUSS;
+      case LSM303_MAG_SCALE2GAUSS:
+        scale = 0x00;
+        break;
+      case LSM303_MAG_SCALE2_5GAUSS:
+        _lsm303_d_mag_config.scale = LSM303_MAG_SCALE4GAUSS;
+      case LSM303_MAG_SCALE4GAUSS:
+        scale = 0x20;
+        break;
+      case LSM303_MAG_SCALE4_7GAUSS:
+        _lsm303_d_mag_config.scale = LSM303_MAG_SCALE8GAUSS;
```

Comment:
If we're just rolling through to the 8 gauss case, that's fine, but don't have multiple assigns to `scale`. Also, the 8 gauss case doesn't assign to `_lsm303_d_mag_config.scale` right now...